pkg/generate/app/cmd/newapp.go

```diff
@@ -1113,6 +1112,25 @@ func (c *AppConfig) run(acceptors app.Acceptors) (*AppResult, error) {
 	}, nil
 }
 
+// checkCircularReferences ensures there are no builds that can trigger themselves
+// due to an imagechangetrigger that matches the output destination of the image.
```

Comment:
Need to document what the argument `objects` is (a "list" of objects that might contain build configs).